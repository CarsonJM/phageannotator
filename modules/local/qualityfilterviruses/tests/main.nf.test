nextflow_process {

    name "Test process: QUALITYFILTERVIRUSES"
    script "../main.nf"
    process "QUALITYFILTERVIRUSES"
    tag "modules"
    tag "modules_local"
    tag "qualityfilterviruses"


    test("['modules_local']['viruses_fna_gz'] & ['modules_local']['proviruses_fna_gz'] & ['modules_local']['quality_summary_tsv']") {

        when {
            params {
                outdir   = "$outputDir"
            }

            process {
                """
                input[0] = [ [ id: 'test' ], [ file(params.test_data['modules_local']['viruses_fna_gz'], checkIfExists: true) ],
                            [ file(params.test_data['modules_local']['proviruses_fna_gz'], checkIfExists: true) ],
                            [ file(params.test_data['modules_local']['quality_summary_tsv'], checkIfExists: true)] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("['modules_local']['viruses_fna_gz'] & ['modules_local']['proviruses_fna_gz'] & ['modules_local']['quality_summary_tsv'] - stub") {

        options "-stub"

        when {
            params {
                outdir   = "$outputDir"
            }

            process {
                """
                input[0] = [ [ id: 'test' ], [ file(params.test_data['modules_local']['viruses_fna_gz'], checkIfExists: true) ],
                            [ file(params.test_data['modules_local']['proviruses_fna_gz'], checkIfExists: true) ],
                            [ file(params.test_data['modules_local']['quality_summary_tsv'], checkIfExists: true)] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
