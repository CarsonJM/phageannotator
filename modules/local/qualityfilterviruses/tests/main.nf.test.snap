{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,2576d8abbe1d5741cca1e890ae4f50a4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4065a8af6c710ebcd076fbcbabc91f17"
                ],
                "filtered_viruses": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,2576d8abbe1d5741cca1e890ae4f50a4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4065a8af6c710ebcd076fbcbabc91f17"
                ]
            }
        ],
        "timestamp": "2023-11-01T14:49:03.483941042"
    }
}